test/unit/kubernetes-deploy/statsd_test.rb

```diff
@@ -40,14 +41,23 @@ def test_build_when_statsd_addr_env_present_but_statsd_implementation_is_not
 
     KubernetesDeploy::StatsD.build
 
-    assert_equal :datadog, StatsD.backend.implementation
+    assert_equal(:datadog, KubernetesDeploy::StatsD.backend.implementation)
   ensure
     ENV['STATSD_ADDR'] = original_addr
     ENV['STATSD_IMPLEMENTATION'] = original_impl
     ENV['STATSD_DEV'] = original_dev
     KubernetesDeploy::StatsD.build
   end
 
+  def test_kubernetes_statsd_does_not_override_global_config
+    KubernetesDeploy::StatsD.build
+    ::StatsD.prefix = "test"
+    ::StatsD.default_sample_rate = 2.0
```

Comment:
Even though it's not that likely other tests use the `::StatsD` class now, we should reset these in an `ensure` since they are global state.